lib/req.ex

```diff
@@ -962,14 +961,10 @@ defmodule Req do
     end
   end
 
-  # TODO: Wait for Finch 0.17
-  @doc false
   def async_request(request, options \\ []) do
```

Comment:
Could you keep all these new function doc false for now? I need to do a pass on their docs anyway.